inference-engine/thirdparty/clDNN/src/gpu/ocl_builder.cpp

```diff
@@ -63,6 +63,9 @@ std::map<std::string, device_impl::ptr> ocl_builder::build_device_list(bool out_
     }
 
     uint32_t idx = 0;
+    cl::Device last_dev;
```

Comment:
Maybe it'd be better to return `std::vector<device_impl::ptr>` from  the `build_device_list*` methods, and then sort and enumerate devices in this vector inside `get_available_devices` method? I think this will be much more readable and all the tricks will be done in a single place.

Reply:
OK, I've changed the approach.